pcf/core/azure_resource.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2019 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pcf.core.particle import Particle
+from azure.common.client_factory import get_client_from_cli_profile
+from azure.mgmt.compute import ComputeManagementClient
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.network import NetworkManagementClient
+from azure.mgmt.storage import StorageManagementClient
+from azure.storage.common import CloudStorageAccount
+
+
+class AzureResource(Particle):
+    """
+    The azure resource class inherits the particle class and handles the clients for azure resources
+    """
+    def __init__(self, particle_definition):
+        super().__init__(particle_definition)
+        self.desired_state_definition = self.particle_definition["azure_resource"]
+        self.custom_config = self.desired_state_definition.get("custom_config", {})
+
+        self._compute_client = None
```

Comment:
think we should only have one client here. the client can default to _resource_client if u want if thats used a lot. otherwise it should be passed into via the particle

Reply:
my justification was just to have multiple clients available in case a particle ever needs one, but that shouldn't ever really happen. I'll change it